src/caf/toolkit/translation.py

```diff
@@ -1060,8 +1071,226 @@ def pandas_multi_vector_zone_translation(
+    matrix_values_column: int | str,
+    translation_from_column: int | str,
+    translation_to_column: int | str,
+    translation_factors_column: int | str,
+) -> None:
+    """Translate zoning system of matrix CSV file.
+
+    Load matrix from CSV, perform translation and write to new
+    CSV. CSV files are expected to be in the matrix 'long' format.
+
+    Parameters
+    ----------
+    matrix_path : pathlib.Path
+        Path to matrix CSV file.
+    translation_path : pathlib.Path
+        Path to translation lookup CSV.
+    output_path : pathlib.Path
+        CSV path to save the translated data to.
+    matrix_zone_columns : tuple[int | str, int | str]
+        Names, or positions, of the 2 columns containing
+        the zone IDs in the matrix file.
+    matrix_values_column : int | str
+        Name, or position, of the column containing the matrix values.
+    translation_from_column : int | str
+        Name, or position, of zone ID column in translation which
+        corresponds to the current vector zone ID.
+    translation_to_column : int | str
+        Name, or position, of column in translation for the new zone IDs.
+    translation_factors_column : int | str
+        Name, or position, of column in translation
+        containing the splitting factors.
+    """
+    # TODO(MB) Handle square format CSVs
+    _validate_column_name_parameters(
+        locals(),
+        "matrix_values_column",
+        "translation_from_column",
+        "translation_to_column",
+        "translation_factors_column",
+    )
+
+    matrix_zone_columns = tuple(matrix_zone_columns)
+    are_strings = any(not isinstance(i, str) for i in matrix_zone_columns)
+    if len(matrix_zone_columns) != 2 or are_strings:
+        raise TypeError(
+            "matrix_zone_columns should be a tuple containing "
+            f"the names of 2 columns not {matrix_zone_columns}"
+        )
+
+    LOG.info("Loading matrix data from: '%s'", matrix_path)
+    matrix = io.read_csv_matrix(
+        matrix_path,
+        format_="long",
+        index_col=matrix_zone_columns,
+        usecols=[*matrix_zone_columns, matrix_values_column],
+        dtype={matrix_values_column: float},
+    )
+    LOG.info(
+        "Loaded matrix with index from '%s' and columns from '%s' containing %s cells",
+        matrix.index.name,
+        matrix.columns.name,
+        matrix.size,
+    )
+
+    lookup, (from_col, to_col, factors_col) = _load_translation(
+        translation_path,
+        translation_from_column,
+        translation_to_column,
+        translation_factors_column,
+    )
+
+    translated = pandas_matrix_zone_translation(
+        matrix,
+        lookup,
+        translation_from_col=from_col,
+        translation_to_col=to_col,
+        translation_factors_col=factors_col,
+    )
+
+    translated.index.name = matrix.index.name
+    translated.columns.name = matrix.columns.name
+    translated.stack().to_csv(output_path)
```

Comment:
Could add in wide or long as options for the output?

Reply:
Added issue #109